plaso/parsers/sqlite_plugins/chrome_history.py

```diff
@@ -27,8 +38,14 @@ def __init__(self):
     """Initializes event data."""
     super(ChromeHistoryFileDownloadedEventData, self).__init__(
         data_type=self.DATA_TYPE)
+    self.danger_type = None
+    self.end_time = None
```

Comment:
let's make these events and represent them with the correct dfDateTime object

Reply:
If we go down that road, what do you think for the attributes then for each event? I see two approaches:

**Option 1**: Make two new events: `chrome:history:file_download:start` and `chrome:history:file_download:end`. Have the `:end` event contain most of the current `chrome:history:file_downloaded` attributes, as one would expect things like `received_bytes`, `state`, `full_path`, and so on to be known for a finished download, not at the start (when no bytes have been downloaded). For the `:start` event, have minimal attributes (maybe just `url` and the implicit timestamp?) 

The issue with this is when a download doesn't complete, it seems the `end_time` value in the SQLite DB is not populated, even though interesting fields like `interrupt_reason` and `state` are.  Since there isn't a timestamp, we couldn't create a `:end` record and we'd miss this info. We could flip and have the `:start` event have all the fields and the `:end` one have just a timestamp, but that seems wrong and confusing (adding future information to an event).

**Option 2**: Make two events, but have their attributes be basically the same. An `:end` record missing means the download didn't finish (and no info is missing, as it's all in the `:start` event). I feel this could be confusing for the user and easy to miss (requires the user to look for the absence of a record - and know to look for it). This also clutters the timeline with near-duplicate events for each download if the download finishes.

I personally favor Option 3: **keep it as one event** (but use dfDateTime objects for `start_time` and `end_time` so it's easier for a person to understand). 

What do you think?

Reply:
Ah, just noticed your other comment on this, re: associated events. I think that would be interesting (for way more than just download events). That isn't built yet though, right? Maybe in the meantime implement Option 2 from above, then 
 "associate" them when that feature is added? 

I'm interested to see how those those events could be shown in Timesketch as well, could be pretty neat.

Reply:
Option 2 is mostly in line what we currently do, associated events is something we could work towards. I opt to brainstorm about this.